core/services/pg/lease_lock.go

```diff
@@ -125,27 +122,34 @@ func (l *leaseLock) TakeAndHold(sig shutdown.Signal) (err error) {
 		isInitial = false
 		l.logRetry(retryCount)
 		retryCount++
+		err = nil
 		select {
-		case <-l.chStop:
-			err = errors.New("stopped")
+		case <-ctx.Done():
+			err = errors.New("stopped by parent context")
+		case <-lctx.Done():
+			err = errors.New("stopped by Release()")
+		case <-time.After(utils.WithJitter(l.refreshInterval)):
+		}
+		if err != nil {
```

Comment:
Resetting the `err` var to nil for reuse read a bit odd to me. What do you think about using something like this closure instead?
```go
err = func() error {
	select {
	case <-ctx.Done():
		return errors.New("stopped by parent context")
	case <-lctx.Done():
		return errors.New("stopped by Release()")
	case <-time.After(utils.WithJitter(l.refreshInterval)):
		return nil
	}
}()
if err != nil {
```